I was able to use the wrapper on iterate decks, so I believe that my Python installation is correct and that my understanding of the wrapper is decent. 

I ran the wrapper, it created the config file, I added my token. And I got the error below (I removed my token)

C:\Users\Hunter\Documents\Misc\Games\Tyrant\tyrant_optimize>fansiteSubmission-master\fansiteSims.py

Could not find config file. Creating one with default settings.

Namespace(config='fansite_config.txt', limit=None, numSims='1000000', numThreads='1', runForever=0, simulator='Tyrant Optimizer', test=0, token='00000000000000000000000000000000')

Traceback (most recent call last):

  File "C:\Users\Hunter\Documents\Misc\Games\Tyrant\tyrant_optimize\fansiteSubmission-master\fansiteSims.py", line 117, in <module>

```
fansiteTest()
```

  File "C:\Users\Hunter\Documents\Misc\Games\Tyrant\tyrant_optimize\fansiteSubmission-master\fansiteSims.py", line 71, in fansiteTest

```
version = simulator.getVersion()
```

  File "C:\Users\Hunter\Documents\Misc\Games\Tyrant\tyrant_optimize\fansiteSubmission-master\fansiteSimulator.py", line 18, in getVersion

```
self.version = self.loadVersion()
```

  File "C:\Users\Hunter\Documents\Misc\Games\Tyrant\tyrant_optimize\fansiteSubmission-master\simulatorTyrantOptimizer.py", line 16, in loadVersion

```
result = subprocess.check_output(commandArgs)
```

  File "C:\Program Files\Python27\lib\subprocess.py", line 544, in check_output

```
raise CalledProcessError(retcode, cmd, output=output)
```

subprocess.CalledProcessError: Command '['tyrant_optimize.exe', '-version']' returned non-zero exit status 4

From: andor9 [mailto:notifications@github.com] 
Sent: Sunday, March 10, 2013 10:27 PM
To: iteratedecks/fansiteSubmission
Subject: [fansiteSubmission] Supoort Tyrant Optimizer (#1)

Also add optional args numThreads, limit and runForever.
All args in config file are passed to addExtraArgs() method to support per-sim-defined args.

---

You can merge this Pull Request by running

  git pull https://github.com/andor9/fansiteSubmission master

Or view, comment on, or merge it at:

  https://github.com/iteratedecks/fansiteSubmission/pull/1

Commit Summary
-   Support Tyrant Optimizer.

File Changes
-   M fansiteConfig.py https://github.com/iteratedecks/fansiteSubmission/pull/1/files#diff-0  (62) 
-   M fansiteHttp.py https://github.com/iteratedecks/fansiteSubmission/pull/1/files#diff-1  (7) 
-   M fansiteSims.py https://github.com/iteratedecks/fansiteSubmission/pull/1/files#diff-2  (75) 
-   M fansiteSimulator.py https://github.com/iteratedecks/fansiteSubmission/pull/1/files#diff-3  (39) 
-   M simulatorIteratedecks.py https://github.com/iteratedecks/fansiteSubmission/pull/1/files#diff-4  (9) 
-   A simulatorTyrantOptimizer.py https://github.com/iteratedecks/fansiteSubmission/pull/1/files#diff-5  (56) 

Patch Links:
-   https://github.com/iteratedecks/fansiteSubmission/pull/1.patch
-   https://github.com/iteratedecks/fansiteSubmission/pull/1.diff

--- a/fansiteSims.py
+++ b/fansiteSims.py
@@ -104,8 +104,10 @@ def fansiteTest():
                     json_data = fansiteHttp.submitSimulation(deckId, sessId, results["total"], results["wins"], results["time"], results["anp"])
                 else:
                     testRepo.testKey(deck, "winrate", 100 * int(results["wins"]) / int(results["total"]))
-                    testRepo.testKey(deck, "drawrate", 100 * int(results["draws"]) / int(results["total"]))
-                    testRepo.testKey(deck, "lossrate", 100 * int(results["losses"]) / int(results["total"]))
+                    if "draws" in results:
+                        testRepo.testKey(deck, "drawrate", 100 * int(results["draws"]) / int(results["total"]))
+                    if "losses" in results:
+                        testRepo.testKey(deck, "lossrate", 100 * int(results["losses"]) / int(results["total"]))
             except NotImplementedError, e:
                 print("Error: Not Implemented: %s" % e)
             except Exception, e:

--- a/fansiteSims.py
+++ b/fansiteSims.py
@@ -90,7 +90,7 @@ def fansiteTest():
             json_data = fansiteHttp.getDecks(sessId, args.limit)
             decks = json_data["decks"]
         else:
-            decks = testRepo.loadTests(True)
+            decks = testRepo.loadTests(False)
         print(" ... " + str(len(decks)) + " decks retrieved")
 
         for deck in decks:

--- a/simulatorTyrantOptimizer.py
+++ b/simulatorTyrantOptimizer.py
@@ -8,10 +8,10 @@ from fansiteSimulator import FansiteSimulator
 class SimulatorTyrantOptimizer(FansiteSimulator):
     name = "Tyrant Optimizer"
     executable = "tyrant_optimize.exe"
-    results_regex = re.compile(r"win%: [.\d]+ \((\d+) out of (\d+)\)"
-        r"\s+draw%: [.\d]+ \((\d+) out of \d+\)"
-        r"\s+loss%: [.\d]+ \((\d+) out of \d+\)"
-        r"\s+ANP: ([\d.]+)")
+    results_regex = re.compile(r"win%: \S+ \((\d+) out of (\d+)\)"
+        r"\s+draw%: \S+ \((\d+) out of \d+\)"
+        r"\s+loss%: \S+ \((\d+) out of \d+\)"
+        r"\s+ANP: (\S+)")
     results_keys = ["wins", "total", "draws", "losses", "anp"]
 
     def loadVersion(self):

--- a/simulatorTyrantOptimizer.py
+++ b/simulatorTyrantOptimizer.py
@@ -8,8 +8,11 @@ from fansiteSimulator import FansiteSimulator
 class SimulatorTyrantOptimizer(FansiteSimulator):
     name = "Tyrant Optimizer"
     executable = "tyrant_optimize.exe"
-    results_regex = re.compile(r"win%: [.\d]+ \((\d+) out of (\d+)\)\D+ANP: ([\d.]+)")
-    results_keys = ["wins", "total", "anp"]
+    results_regex = re.compile(r"win%: [.\d]+ \((\d+) out of (\d+)\)"
+        r"\s+draw%: [.\d]+ \((\d+) out of \d+\)"
+        r"\s+loss%: [.\d]+ \((\d+) out of \d+\)"
+        r"\s+ANP: ([\d.]+)")
+    results_keys = ["wins", "total", "draws", "losses", "anp"]
 
     def loadVersion(self):
         commandArgs = [self.executable, "-version"]
@@ -37,7 +40,7 @@ class SimulatorTyrantOptimizer(FansiteSimulator):
 
     def addExtraArgs(self, commandArgs, args):
         commandArgs.extend(["-t", str(getattr(args, "numThreads", 1))])
-        commandArgs.extend(["sim", args.numSims])
+        commandArgs.extend(["sim", str(args.numSims)])
 
     def addMission(self, commandArgs, missionId):
         commandArgs.append("Mission #%s" % missionId)

--- a/testRepo.py
+++ b/testRepo.py
@@ -24,19 +24,13 @@ def updateTests(dataDirectory = "", filename = "tests.json", host = "haileon.com
 def loadTests(forceUpdate = False, dataDirectory = "", filename = "tests.json", host = "haileon.com", path = "/SimTyrantJS_ListTestCases"):
     data = None
     if(not forceUpdate):
-        fileDest = dataDirectory + filename
-        f = None
-        try:
-            f = open(fileDest, 'rb')
+        fileDest = os.path.join(dataDirectory, filename)
+        with open(fileDest, 'rb') as f:
             contents = f.read()
-        finally:
-            f.close()
         if(data is None):
             forceUpdate = True
-
     if(forceUpdate):
         data = updateTests(dataDirectory, filename, host, path)
-
     json_data = json.loads(data)
     decks = json_data["decks"]
     return decks

--- a/testRepo.py
+++ b/testRepo.py
@@ -25,11 +25,10 @@ def loadTests(forceUpdate = False, dataDirectory = "", filename = "tests.json", 
     data = None
     if(not forceUpdate):
         fileDest = os.path.join(dataDirectory, filename)
-        with open(fileDest, 'rb') as f:
-            contents = f.read()
-        if(data is None):
-            forceUpdate = True
-    if(forceUpdate):
+        if(os.path.exists(fileDest)):
+            with open(fileDest, 'rb') as f:
+                data = f.read()
+    if(data is None):
         data = updateTests(dataDirectory, filename, host, path)
     json_data = json.loads(data)
     decks = json_data["decks"]

--- a/tyrantData.py
+++ b/tyrantData.py
@@ -4,31 +4,21 @@ import os
 
 def getCheck(dataDirectory = ""):
     check = {}
-
     files = ["achievements", "cards", "missions", "quests", "raids"]
-
-    for file in files:
-        path = dataDirectory + file + ".xml"
-        hash = None
-        if(os.path.exists(path)):
-            contents = None
-            f = None
-            try:
-                f = open(path, 'rb')
+    for filename in files:
+        path = os.path.join(dataDirectory, filename + ".xml")
+        if os.path.exists(path):
+            with open(path, 'rb') as f:
                 contents = f.read()
-            finally:
-                f.close()
-            hash = hashlib.md5(contents).hexdigest()
-            check[file] = hash
+            md5hash = hashlib.md5(contents).hexdigest()
+            check[filename] = md5hash
     return check
 
 def updateDataFiles(source = "kg.tyrantonline.com", dataDirectory = "", files = ["achievements", "cards", "missions", "quests", "raids"]):
     print("Getting data from " + source)
-
-    for file in files:
-
-        fileSource = "/assets/" + file + ".xml"
-        fileDest = dataDirectory + file + ".xml"
+    for filename in files:
+        fileSource = "/assets/" + filename + ".xml"
+        fileDest = dataDirectory + filename + ".xml"
         http = httplib.HTTPConnection(source, 80, timeout=10)
         http.request("GET", fileSource)
 

